docs/source/integrations/deltalake_versioning.md

```diff
@@ -0,0 +1,127 @@
+
+When you run the Kedro project with `kedro run` command, the Delta table will be saved to the location specified in the `filepath` argument as a folder of `parquet` files. This folder also contains a `_delta_log` directory which stores the transaction log of the Delta table. The following runs of the pipeline will create new versions of the Delta table in the same location and new entries in the `_delta_log` directory.
+
+Suppose the upstream datasets `companies`, `shuttles`, or `reviews` is updated. You can run the following command to generate a new version of the `model_input_table` dataset:
+
+```bash
+kedro run --to-outputs=model_input_table
+```
+To inspect the updated dataset and logs:
+```bash
+tree data/03_primary
+>> data/03_primary
+└── model_input_table
+    ├── _delta_log
+    │   ├── 00000000000000000000.json
+    │   ├── 00000000000000000001.json
+    ├── part-00001-73cbb76d-88e8-4a58-a8f1-4a87b5283203-c000.snappy.parquet
+    ├── part-00001-7498e001-81e7-4098-b651-8ae4f6e844c9-c000.snappy.parquet
+    └── part-00001-9b79408f-b5cb-4400-9f26-103ab28da96c-c000.snappy.parquet
+```
+### Load a specific version of the dataset
+To load a specific version of the dataset, you can specify the version number in the `load_args` parameter in the catalog entry:
+
+```yaml
+model_input_table:
+  type: pandas.DeltaTableDataset
+  filepath: data/03_primary/model_input_table
+  load_args:
+    version: 1
+```
+
+## Inspect the dataset in interactive mode
+
+You can inspect the history and the metadata of the Delta table in an interactive Python session. To start the IPython session with Kedro components loaded, run:
+
+```bash
+kedro ipython
+```
+
+You can load the Delta table using the `catalog.load` method and inspect the dataset:
+
+```python
+model_input_table = catalog.datasets['model_input_table']
+```
+You can inspect the history of the Delta table by accessing the `history` attribute:
+```python
+>> model_input_table.history
+
+[Out]: [
+    {
+        'timestamp': 1739891304488,
+        'operation': 'WRITE',
+        'operationParameters': {'mode': 'Overwrite'},
+        'operationMetrics': {
+            'execution_time_ms': 8,
+            'num_added_files': 1,
+            'num_added_rows': 6027,
+            'num_partitions': 0,
+            'num_removed_files': 1
+        },
+        'clientVersion': 'delta-rs.0.23.1',
+        'version': 1
+    },
+    {
+        'timestamp': 1739891277424,
+        'operation': 'WRITE',
+        'operationParameters': {'mode': 'Overwrite'},
+        'clientVersion': 'delta-rs.0.23.1',
+        'operationMetrics': {
+            'execution_time_ms': 48,
+            'num_added_files': 1,
+            'num_added_rows': 6027,
+            'num_partitions': 0,
+            'num_removed_files': 0
+        },
+        'version': 0
+    }
+]
+```
+
+You can also inspect the loaded version of the table with the following method:
+
+```python
+>> model_input_table.get_loaded_version()
+[Out]: 1
```

Comment:
Or maybe your Python REPL is like this? 🤔 not sure if this is the normal `python` REPL, IPython, or something else